include/posix/time.h

```diff
@@ -10,15 +10,29 @@
 extern "C" {
 #endif
 
+
+#ifndef __timespec_defined
+#define __timespec_defined
 struct timespec {
-	signed int  tv_sec;
-	signed int  tv_nsec;
+	signed int tv_sec;
+	signed int tv_nsec;
 };
+#endif
 
+/* Older newlib's like 2.{0-2}.0 don't define any newlib version defines, only
```

Comment:
@galak, The proper fix for this would be the same as for mode_t - figure out the proper header belongs to, rely on newlib's existing definition there, and move def from posix/* to minimal/*. (Again, when we repeat that process a hundred of times, we'll find that we just recreated newlib headers in minimal. So, it's still not to late to say "POSIX and minimal don't plat well together, a [decent] libc is part of POSIX", and concentrate only on POSIX+newlib).